src/include/storage/page/page_guard.h

```diff
@@ -59,6 +61,26 @@ class BasicPageGuard {
    */
   ~BasicPageGuard();
 
+  /** TODO(P2): Add implementation
+   *
+   * @brief Upgrade a BasicPageGuard to a ReadPageGuard
+   *
+   * The protected page is not evicted from the buffer pool during the upgrade.
+   *
+   * @return an upgraded ReadPageGuard
+   */
+  auto UpgradeRead() -> ReadPageGuard;
```

Comment:
should tell students to make the current page guard invalid after calling this function.

Reply:
(and we should have test cases on that?)